GetStream/stream-chat-android
# [3156] Fix attachment issue and message sending issues

Closes #3156 

### 🎯 Goal
There were a few bugs that happen when the user is changed and we send messages. One of the bug was that the attachments if not uploaded completely and the user is changed, then the attachment message is sent from the new user instead of the original user who created this message. For more info -> https://github.com/GetStream/stream-chat-android/issues/3156

Another bug was when you change the user in the sample app and try to send an attachment, attachments are not sent and got stuck. 

### 🛠 Implementation details
The problem was happening as we were not clearing the interceptors on user disconnect. So a new `removeAllInterceptors` method is added to `ChatClient` to clear interceptors when a user is disconnected. 

### 🧪 Testing
#### Testing first bug:
1. Login with any user (let's say user amit) in the sample app,
2. Go into any channel, Upload an attachment (make sure the attachment is big enough so we can do step number 3,4, etc).
3. While the attachments are being uploaded, switch back to another user (for ex. Jc) (simply sign out and log in to another user).
4. Open the same channel, attachments should not either upload  or if uploaded it should be from the original message author account.

#### Testing second bug:
1 - Log in to an account
2 - Log out
3 - Log in with another account
4 - Try to send attachments, attachments should be sent and not stuck at 0kb.

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
